### ISSUE TITLE
Ensuring of extern function gets silently dropped during extraction

### ISSUE THREAD

**Issue by jad-hamza** [epfl-lara/stainless#549]
```scala
import stainless.annotation._

object EnsuringExtraction {
  @extern def f(x: BigInt, y: BigInt): Unit = {
    require(x >= 0)
    (??? : Unit).ensuring(_ => y >= 0)
  }
}
```

This is extracted as:

```scala
@extern
def f(x: BigInt, y: BigInt): Unit = {
  require(x >= 0)
   <empty tree>[Unit]
}
```

Extraction works fine when `ensuring` is outside or when we remove the `@extern` annotation.

**Comment by jad-hamza** [comment]
The problem here is that `ensuring` is part of the body of the extern function `f`, and is therefore dropped as any other expression. I'm not sure what's the best way to report this. We could emit a warning when we drop a `require/ensuring/assert` (which are a bit Stainless keywords) from an `@extern` function. What do you think?

**Comment by romac** [comment]
Closed via #549
